tests/test_git_analyzer.sh

```
@@ -0,0 +1,241 @@
+    local test_repo_dir
+    test_repo_dir=$(mktemp -d)
+    
+    # 現在のディレクトリを保存
+    pushd "$test_repo_dir" >/dev/null
+    
+    git init >/dev/null 2>&1
+    git config user.email "test@example.com"
+    git config user.name "Test User"
+    
+    # クリーンアップ用の関数を設定
+    trap "popd >/dev/null 2>&1; rm -rf '$test_repo_dir'" EXIT
```

Shellcheck (0.10.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)